deeplake/core/transform/transform_tensor.py

```diff
@@ -136,7 +136,10 @@ def append(self, item):
                 self._ndim = len(shape)
             else:
                 if len(shape) != self._ndim:
-                    raise TensorInvalidSampleShapeError(shape, self._ndim)
+                    if not (
+                        isinstance(item, Sample) and len(shape) == 2 and self._ndim == 3
```

Comment:
we should also check if its image

Reply:
we can't check, it could be a sample created from array as well, not necessarily with a path or shape

Reply:
is there any reason to have two if statements? It is not critical, just curious.

Reply:
not really, putting everything in 1 line was making it hard to read